src/core/src/main/java/org/apache/jmeter/testelement/TestPlan.java

```diff
@@ -82,12 +85,23 @@ public void prepareForPreCompile()
     }
 
     /**
-     * Fetches the functional mode property
-     *
+     * Fetches the functional mode property<br>
+     * Could be change for no-GUI test with jmeter property: {@code PROP_FUNCTIONAL_MODE}
      * @return functional mode
      */
     public boolean isFunctionalMode() {
-        return getPropertyAsBoolean(FUNCTIONAL_MODE);
+        boolean functionalModeDefault = getPropertyAsBoolean(FUNCTIONAL_MODE);
+        log.debug("functionalModeDefault=" + functionalModeDefault);
```

Comment:
Please either remove the logging or use patterns for log formatting
```suggestion
        log.debug("functionalModeDefault={}", functionalModeDefault);
```